tt_metal/impl/device/device.hpp

```diff
@@ -341,29 +304,67 @@ class Device {
     std::tuple<SubDeviceManagerId, SubDeviceId> create_sub_device_manager_with_fabric(tt::stl::Span<const SubDevice> sub_devices, DeviceAddr local_l1_size);
     std::optional<SubDeviceId> get_fabric_sub_device_id() const;
 
-    program_cache::detail::ProgramCache& get_program_cache() { return program_cache_; }
-
+    // review: only used by transpose and matmul
     std::set<CoreCoord> get_compute_cores() const { return compute_cores_; }
```

Comment:
The reach-arounds in transpose and matmul should have been removed instead of adding this function in the previous pr. Matmul and transpose can call `device->lowest_occupied_compute_l1_address()` instead of doing `const std::vector<uint32_t>& bank_ids =
        device->bank_ids_from_logical_core(BufferType::L1, *device->get_compute_cores().begin());
    std::optional<uint64_t> lowest_address =
        allocator::lowest_occupied_l1_address(*device->get_initialized_allocator(), bank_ids[0]);`

Reply:
Great! Will replace and remove the method.